packages/modular-scripts/src/build-scripts/webpack-scripts/startWebpack.ts

```diff
@@ -0,0 +1,130 @@
+import { choosePort } from '../common-scripts/getPort';
+import prepareUrls from '../common-scripts/urls';
+import { Paths } from '../common-scripts/determineTargetPaths';
+import openBrowser from '../common-scripts/openBrowser';
+
+// Tools like Cloud9 rely on this.
+const portEnv = process.env.PORT;
+const DEFAULT_PORT = portEnv ? parseInt(portEnv, 10) : 3000;
+const HOST = process.env.HOST || '0.0.0.0';
+
+export default function startWebpack(
+  esbuildTargetFactory: string[],
+  isApp: boolean,
+  dependencyMap: Map<string, string>,
+  useReactCreateRoot: boolean,
+  styleImports: Set<string>,
+  paths: Paths,
+) {
+  if (process.env.HOST) {
+    log(
+      chalk.cyan(
+        `Attempting to bind to HOST environment variable: ${chalk.yellow(
+          chalk.bold(process.env.HOST),
+        )}`,
+      ),
+    );
+    log(
+      `If this was unintentional, check that you haven't mistakenly set it in your shell.`,
+    );
+    log(`Learn more here: ${chalk.yellow('https://cra.link/advanced-config')}`);
+    log();
+  }
+
+  choosePort(HOST, DEFAULT_PORT)
+    .then(async (port: number | undefined) => {
+      if (!port) {
+        // We have not found a port.
+        return;
+      }
+
+      const config = await getConfig(
+        false,
+        esbuildTargetFactory,
+        isApp,
+        dependencyMap,
+        useReactCreateRoot,
+        styleImports,
+        paths,
+      );
+      // overload for webpack-dev-server@4
+      config.stats = 'none';
+      config.infrastructureLogging = {
+        level: 'none',
+      };
+
+      const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
+      const packageJson = readJSONSync(
+        require.resolve(paths.appPackageJson),
+      ) as PackageJson;
+      const appName = packageJson.name;
+      const useTypeScript = !isCI && fs.existsSync(paths.appTsConfig);
+      const urls = prepareUrls(
+        protocol,
+        HOST,
+        port,
+        paths.publicUrlOrPath.slice(0, -1),
+      );
+      // Create a webpack compiler that is configured with custom messages.
+      // Only run typecheck if not in CI env
+      const compiler = createCompiler(appName, config, urls, useTypeScript);
+
+      // Load proxy config
+      const proxySetting = packageJson.proxy as unknown;
+      const proxyConfig = prepareProxy(
+        proxySetting,
+        paths.appPublic,
+        paths.publicUrlOrPath,
+      );
+      // Serve webpack assets generated by the compiler over a web server.
+      const serverConfig = createDevServerConfig(
+        port,
+        proxyConfig,
+        urls.lanUrlForConfig,
+        paths,
+      );
+
+      // Types from WebpackDevServer and Webpack overlap but unfortunately aren't compatible
+      // even though when you look into it Webpack-Dev-Server simply imports the Webpack types
+      const devServer = new WebpackDevServer(
+        serverConfig,
+        compiler as unknown as Compiler,
```

Comment:
TypeScript doesn't manage to figure out that the Compiler Type from Webpack and that from Webpack-Dev-Server are the same, and fails to compare the two as they are 'too big' to compare and assert that they are the same.